api/src/test/java/com/epam/pipeline/security/saml/SAMLUserDetailsServiceImplTest.java

```diff
@@ -16,86 +16,97 @@
+import static com.epam.pipeline.security.saml.SamlUserRegisterStrategy.EXPLICIT;
+import static com.epam.pipeline.security.saml.SamlUserRegisterStrategy.EXPLICIT_GROUP;
+import static com.epam.pipeline.util.CustomAssertions.assertThrows;
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.anyListOf;
+import static org.mockito.Matchers.anyLong;
+import static org.mockito.Matchers.anyMapOf;
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Matchers.eq;
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.when;
+
+public class SAMLUserDetailsServiceImplTest {
 
     private static final String USER_NAME = "TEST_USER";
+    private static final String TEST_STRING = "TEST";
     private static final String OLD_USER_NAME = "user_name";
     private static final String SAML_ATTRIBUTE_1 = "ATTR_1";
     private static final String SAML_ATTRIBUTE_2 = "ATTR_2";
     private static final String SAML_ATTRIBUTES_STRING = "ATTR_3";
     private static final String ATTRIBUTES_KEY_1 = "email";
     private static final String ATTRIBUTES_KEY_2 = "user";
-    private static final String TEST_FOLDER = "test-folder";
-    private PipelineUser user = new PipelineUser();
-    private Map<String, String> expectedAttributes = new HashMap<>();
-    private List<String> groups = Stream.of(SAML_ATTRIBUTE_1, SAML_ATTRIBUTE_2).collect(Collectors.toList());
-    private UserContext expectedUserContext = new UserContext(1L, USER_NAME.toUpperCase());
     private static final String SAML_ATTRIBUTE_BLOCKED_USER_VALUE_1 = "true";
     private static final String SAML_ATTRIBUTE_BLOCKED_USER_VALUE_2 = "True";
     private static final String SAML_ATTRIBUTE_BLOCKED_USER_VALUE_3 = "TRUE";
+    private static final String SAML_ATTRIBUTE_BLOCKED_USER_VALUE_4 = "tRUe";
     private static final String SAML_ATTRIBUTE_NOT_BLOCKED_USER_VALUE = "false";
 
-    @InjectMocks
-    @Autowired
-    private SAMLUserDetailsServiceImpl userDetailsService;
-    @Autowired
-    private GrantPermissionManager grantPermissionManager;
-    @Autowired
-    private FolderManager folderManager;
+    private final PipelineUser user = new PipelineUser();
+    private final List<String> groups = Stream.of(SAML_ATTRIBUTE_1, SAML_ATTRIBUTE_2).collect(Collectors.toList());
+    private final UserContext expectedUserContext = new UserContext(1L, USER_NAME.toUpperCase());
+    private Map<String, String> expectedAttributes = new HashMap<>();
 
     @Mock
-    private SAMLCredential credential;
+    private GrantPermissionManager mockPermissionManager;
 
     @Mock
-    private NameID nameID;
+    private SAMLCredential mockCredential;
 
-    @MockBean
-    private UserManager userManager;
+    @Mock
+    private NameID mockNameID;
+
+    @Mock
+    private UserManager mockUserManager;
+
+    @Mock
+    private MessageHelper mockMessageHelper;
```

Comment:
Let's add `@SuppressWarnings("PMD.UnusedPrivateField")` to avoid `pmdTest` task failure.